Sources/TuistGenerator/Generator/WorkspaceGenerator.swift

```diff
@@ -116,13 +117,56 @@ final class WorkspaceGenerator: WorkspaceGenerating {
 
         try write(xcworkspace: xcWorkspace, to: workspacePath)
 
+        // SPM
+
+        try generatePackageDependencyManager(at: path,
+                                             workspace: workspace,
+                                             workspaceName: workspaceName,
+                                             graph: graph)
+
         // CocoaPods
 
         try cocoapodsInteractor.install(graph: graph)
 
         return workspacePath
     }
 
+    private func generatePackageDependencyManager(
+        at path: AbsolutePath,
+        workspace _: Workspace,
+        workspaceName: String,
+        graph: Graphing
+    ) throws {
+        let packages = try graph.targets.flatMap { try graph.packages(path: $0.path, name: $0.name) }
+
+        if packages.isEmpty {
+            return
+        }
+
+        let hasRemotePackage = packages.first(where: { package in
+            switch package.packageType {
+            case .remote: return true
+            case .local: return false
+            }
+        }) != nil
+
+        let rootPackageResolvedPath = path.appending(component: "Package.resolved")
+        let workspacePackageResolvedPath = path.appending(RelativePath("\(workspaceName)/xcshareddata/swiftpm/Package.resolved"))
+
+        let workspacePath = path.appending(component: workspaceName)
+        // -list parameter is a workaround to resolve package dependencies for given workspace without specifying scheme
+        try system.runAndPrint(["xcodebuild", "-resolvePackageDependencies", "-workspace", workspacePath.pathString, "-list"])
+
+        if hasRemotePackage {
+            if FileHandler.shared.exists(rootPackageResolvedPath) {
```

Comment:
Aren't we defeating the purpose of the `Package.resolved` file? If we delete the one existing in the root directory, we'll be resolving dependencies every time the project gets generated. Let me know if I misunderstood anything. This is how I see the process:

1. Copy the `Package.resolved` from the root directory into the workspace.
2. Resolve the dependencies using `xcodebuild`
3. Copy the `Package.resolved` path back to the root directory.

Reply:
Looks like this is run after and the file is then linked

Reply:
You're right that we should try to copy `.package.resolved` there first, so I'd delete everything after `system.runAndPrint()` and write it like this:
```swift
if hasRemotePackage {
            if FileHandler.shared.exists(workspacePackageResolvedPath) {
                try FileHandler.shared.delete(workspacePackageResolvedPath)
            }
            
            if !FileHandler.shared.exists(rootPackageResolvedPath) {
                FileHandler.shared.touch(rootPackageResolvedPath)
            }
            
            FileHandler.shared.linkFile(atPath: rootPackageResolvedPath, toPath: workspacePackageResolvedPath)
        }

        let workspacePath = path.appending(component: workspaceName)
        // -list parameter is a workaround to resolve package dependencies for given workspace without specifying scheme
        try system.runAndPrint(["xcodebuild", "-resolvePackageDependencies", "-workspace", workspacePath.pathString, "-list"])
```

So, the process would be like this:
1. Delete workspace `Package.resolved` if it exists
2. Touch root package if it does not exist
3. Link workspace package to root package
4. Resolve packages -> if `.package.resolved` was empty, this should update it alongside with it

Should I incorporate the change like this or is there something I'm missing?

Reply:
Also now that I think about it, I will have to create intermediate directories for workspace package, otherwise that probably would not work. But I'd add additional tests to cover that.

Reply:
> Should I incorporate the change like this or is there something I'm missing?

That sounds perfect.

> Also now that I think about it, I will have to create intermediate directories for workspace package, otherwise that probably would not work. But I'd add additional tests to cover that.

Yeah, you'll need to create intermediate directories, otherwise the creation of some of the files might fail.

Reply:
Logic seems sound to me @fortmarek!

Reply:
Had to change it a little bit, so check it out (this could not work because I was copying empty `Package.resolved` to workspace, which would crash on the error "Malformed package") - but the solution is still quite similar.